.github/workflows/labeler.yml

```diff
@@ -53,12 +53,14 @@ jobs:
         env:
           event_json: "${{ toJSON(github.event) }}"
           GITHUB_TOKEN: ${{ steps.create_token.outputs.token }}
-        run:
-          ./venv/bin/python hacking/pr_labeler/label.py issue ${{ github.event.issue.number || inputs.number }}
+          number: "${{ github.event.issue.number || inputs.number }}"
+        run: |
+          ./venv/bin/python hacking/pr_labeler/label.py issue "${number}"
       - name: "Run the PR labeler"
         if: "github.event.pull_request || inputs.type == 'pr'"
         env:
           event_json: "${{ toJSON(github.event) }}"
           GITHUB_TOKEN: ${{ steps.create_token.outputs.token }}
-        run:
-          ./venv/bin/python hacking/pr_labeler/label.py pr ${{ github.event.number || inputs.number }}
+          number: "${{ github.event.number || inputs.number }}"
+        run: |
+          ./venv/bin/python hacking/pr_labeler/label.py pr "${number}"
```

Comment:
In addition to be comment above, I'd suggest only using single quotes as with double quotes, it's possible to inject subshell execution. As in `"$(/bin/sh evil-command-works)"` vs. `'$(/bin/sh evil-command-is-harmless)'` — only the former is vulnerable because bash performs string interpolation there.

Reply:
`'${number}'` would expand verbatim to `${number}`, which is not what we want. As for quoting the `${{ }}` GHA templating expansions when used directly in the script body, you can always escape the quotes, so it doesn't really matter from a security perspective, although again, this is a trusted input in the first place.

Reply:
Okay, makes sense.

Reply:
Marking this as resolved.